orttraining/orttraining/core/optimizer/memory_optimizer/memory_insight.h

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/common/inlined_containers.h"
+#include "orttraining/core/optimizer/memory_optimizer/common.h"
+#include "orttraining/core/optimizer/memory_optimizer/optimization_planner.h"
+#include "orttraining/core/optimizer/memory_optimizer/recompute_analysis.h"
+
+namespace onnxruntime::optimizer::memory_optimizer {
+
+/**
+ * @brief A data structure to store memory optimization statistics for a specific node cluster id.
+ *
+ * We will collect statistics for each node cluster id.
+ * The node cluster id is generated from all possible optimization plans for a specific node, plus shape, data type,
+ * outputs, etc. For the nodes have the same node cluster id, they will have one single MemoryRecord, displayed
+ * as a row in the final memory optimization statistics table.
+ */
+class MemoryRecord {
+ public:
+  // Recompute Column
+  std::string recompute_subgraph_str;
+  /// output index, shape, byte count per element
+  std::vector<std::tuple<size_t, std::string, int>> recomputed_outputs;
```

Comment:
>std::tuple<size_t, std::string, int> [](http://example.com/codeflow?start=14&length=36)

Would a structure be a more meaningful representation with named members?

Reply:
defined `MemoryRecord::RecomputeStat` for it. Thanks!